tools/sqldatabase/sql_database.go

```diff
@@ -0,0 +1,154 @@
+package sqldatabase
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+)
+
+// EngineFunc is the function that returns the database engine.
+type EngineFunc func(string) (Engine, error)
+
+//nolint:gochecknoglobals
+var engines = make(map[string]EngineFunc)
+
+func RegisterEngine(name string, engineFunc EngineFunc) {
+	engines[name] = engineFunc
+}
+
+// Engine is the interface that wraps the database.
+type Engine interface {
+	// Dialect returns the dialect(e.g. mysql, sqlite, postgre) of the database.
+	Dialect() string
+
+	// Query executes the query and returns the columns and results.
+	Query(ctx context.Context, query string, args ...any) (cols []string, results [][]string, err error)
+
+	// TableNames returns all the table names of the database.
+	TableNames(ctx context.Context) ([]string, error)
+
+	// TableInfo returns the table information of the database.
+	// Typically, it returns the CREATE TABLE statement.
+	TableInfo(ctx context.Context, tables string) (string, error)
```

Comment:
it might be nice to make tables variadic and output all if omitted

Reply:
Yes, the fist version of the parameter tables is a string slice。 However, considering it may be called concurrently by their way.